src/qml/UnitsPage.qml

```diff
@@ -38,40 +34,38 @@ Item {
         defaultValue: false
     }
 
-    GridLayout {
-        columns: 2
+    property string rowHeight: Dims.h(25)
+
+    Column {
         anchors.fill: parent
-        anchors.margins: Dims.l(15)
 
-        Label {
+        Item { width: parent.width; height: Dims.l(25) }
+
+        Item {
+            width: parent.width
+            height: rowHeight
+        LabeledSwitch {
+            anchors.fill: parent
+            height: rowHeight
             //% "Use 12H format:"
             text: qsTrId("id-12h-format")
-            font.pixelSize: Dims.l(6)
-            verticalAlignment: Text.AlignVCenter
-            wrapMode: Text.Wrap
-            Layout.maximumWidth: Dims.w(50)
-        }
-
-        Switch {
-            Component.onCompleted: checked = use12H.value
+            checked: use12H.value
             onCheckedChanged: use12H.value = checked
-            width: Dims.l(20)
         }
+        }
+        Item {
+            width: parent.width
+            height: rowHeight
 
-        Label {
+        LabeledSwitch {
```

Comment:
If this LabeledSwitch is under the Item then it should also get an extra indentation